models/articles.model.js

```diff
@@ -11,52 +11,61 @@ function fetchArticleId(article_id) {
     });
 }
 
-function fetchAllArticles() {
-    return db
-      .query(
-        `SELECT 
-    articles.article_id, 
-    articles.title, 
-    articles.topic, 
-    articles.author,  
-    articles.created_at,
-    articles.votes,
-    articles.article_img_url,
-    COUNT(comments.comment_id)::INTEGER AS comment_count
-    FROM articles
-    LEFT JOIN comments ON articles.article_id = comments.article_id
-    GROUP BY articles.article_id
-    ORDER BY articles.created_at DESC;`
-      )
-      .then(({ rows: articles }) => {
-        return articles;
-      });
+function fetchAllArticles(topic) {
+  let sqlQuery = `SELECT 
+  articles.article_id, 
+  articles.title, 
+  articles.topic, 
+  articles.author,  
+  articles.created_at,
+  articles.votes,
+  articles.article_img_url,
+  COUNT(comments.comment_id)::INTEGER AS comment_count
+  FROM articles
+  LEFT JOIN comments ON articles.article_id = comments.article_id`;
+
+  const articleTopic = [];
+  if (topic) {
+    const topicQuery = ` WHERE articles.topic=$1`;
+    sqlQuery += topicQuery;
+    articleTopic.push(topic);
+  }
+
+  sqlQuery += ` GROUP BY articles.article_id`;
 
+  return db.query(sqlQuery, articleTopic)
+  .then(({ rows: articles }) => {
+    if (articles.length === 0) {
```

Comment:
Will need to change this slightly as if a topic exists but has no articles, it would need to return an empty array and a 200 rather than a 404
To overcome this, we'll need to check if a topic exists or not - remember you have an endpoint that does this 👀